### PR TITLE
`Learning analytics`: Add judgement of learning component

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [ ] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

We want to see how students judge themselves with regards to the mastery of a competency.

### Description

- Remove lots of bugs in star rating and make the fractional stars work
- Add a judgement of learning component -> Placed it on the student course dashboard for testing

### Steps for Testing
Navigate to the student course dashboard and check it out

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Screenshot 2024-05-15 at 14 59 59
Screenshot 2024-05-15 at 15 00 12

### CODE REVIEW

**Comment by FelixTJDietrich** [comment]
Somehow I did not add the JoL component, it is committed now...

File: src/main/webapp/app/overview/course-dashboard/judgement-of-learning-rating/judgement-of-learning-rating.component.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * Create new master rating for competency and receive the artemis rating
+     * @param event - starRating component that holds new rating value
+     */
+    onRate(event: { oldValue: number; newValue: number; starRating: StarRatingComponent }) {
+        if (this.rating !== undefined) {
+            return;
+        }
+
+        this.rating = event.newValue;
+        this.artemisRating = Math.random() * 5;
```
Comment (kaancayli) [inline src/main/webapp/app/overview/course-dashboard/judgement-of-learning-rating/judgement-of-learning-rating.component.ts:26]: I guess, `this.artemisRating` still uses mock data

File: src/main/webapp/app/overview/course-dashboard/course-dashboard.component.html [hunk 2]
```diff
@@ -15,4 +15,5 @@ <h4 jhiTranslate="artemisApp.courseStudentDashboard.overallPerformance.title"></
     } @else {
         <div jhiTranslate="artemisApp.courseStudentDashboard.noMetrics"></div>
     }
+    <jhi-judgement-of-learning-rating />
```
Comment (kaancayli) [inline src/main/webapp/app/overview/course-dashboard/course-dashboard.component.html:18]: This should be placed inside a competency I assume.

**Review by kaancayli — APPROVED** [review]
I have made some comments. These can however be addressed later, once everything comes together. Aside those comments, code looks good. Approved 👍🏻

**Pull request merged**